include/deal.II/base/vectorization.h

```diff
@@ -68,6 +68,27 @@ namespace std
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  template <typename T>
+  struct NumberType 
+  {
+    static T value (const T &t) { return t; }
+  };
```

Comment:
Could you please add a short description at the top of this struct that summarizes what we have discussed here. It should contain why we need this construction.